DOCUMENTATION.md

```diff
@@ -2099,68 +2093,8 @@ hz.getReliableTopic('my-distributed-topic').then(function (t) {
 
 Hazelcast Reliable Topic uses `MessageListener` to listen to the events that occur when a message is received. See the [Message Listener section](#7524-message-listener) for information on how to create a message listener object and register it.
 
-## 7.4.9. Using Lock
```

Comment:
CP Subsystem is in the scope of 4.0. I thought we can add the CP versions of the removed sections when we implement it.

Reply:
Yes, it's in scope, but once this PR is merged, users will see the readme available on `master` branch and will have questions about old non-CP DSes. To avoid those questions we should either keep the documentation consistent, or create a new branch (`4.0-staging` or `4.0-rc`) and merge this and other 4.0 PRs there. Once the 4.0 client will be feature complete, we can merge this new branch into `master`. WDYT?

Reply:
Got it, added a new section about them

Reply:
As it was discussed on Slack recently, we should target this PR against `master`, but add the following disclaimer on top of readme:
```
This client is under heavy development for version 4.0.

Please switch to a 3.12.x branch or tag if you are using a 3.12.x version of Hazelcast IMDG.
```

P.S. We should be using `3.12.z` name instead of `3.12.x` as it seems to be the company-wide practice. So, it makes sense to rename the branch here and in git.

Reply:
Done. If it is okay, I will do the renaming after merging this PR